cmd/kots/cli/velero.go

```diff
@@ -77,12 +87,484 @@ func VeleroEnsurePermissionsCmd() *cobra.Command {
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			v := viper.GetViper()
+
+			clientset, err := k8sutil.GetClientset()
+			if err != nil {
+				return errors.Wrap(err, "failed to get clientset")
+			}
+
+			if !kotsutil.IsKurl(clientset) {
+				return errors.New("configuring snapshots to use the internal store is only supported for embedded clusters")
+			}
+
+			namespace := metav1.NamespaceDefault
+
+			veleroStatus, err := snapshot.DetectVelero(cmd.Context(), namespace)
+			if err != nil {
+				return errors.Wrap(err, "failed to detect velero")
+			}
+			if veleroStatus == nil {
+				return errors.New("velero namespace not found")
+			}
+
+			if !veleroStatus.ContainsPlugin("velero-plugin-for-aws") {
+				return errors.New("velero does not have the 'velero-plugin-for-aws' installed; " +
+					"consult https://kots.io/kotsadm/snapshots/overview/ for install instructions`)")
+			}
+
+			registryOptions, err := kotsadm.GetKotsadmOptionsFromCluster(namespace, clientset)
+			if err != nil {
+				return errors.Wrap(err, "failed to get registry options from cluster")
+			}
+
+			configureStoreOptions := snapshot.ConfigureStoreOptions{
+				Internal:          true,
+				KotsadmNamespace:  namespace,
+				RegistryOptions:   &registryOptions,
+				SkipValidation:    v.GetBool("skip-validation"),
+				ValidateUsingAPod: true,
+			}
+			_, err = snapshot.ConfigureStore(cmd.Context(), configureStoreOptions)
+			if err != nil {
+				return errors.Wrap(err, "failed to configure store")
+			}
+
+			log := logger.NewCLILogger()
+			log.Info("\nStore Configured Successfully")
+
+			return nil
+		},
+	}
+
+	cmd.Flags().Bool("skip-validation", false, "skip the validation of the internal store endpoint/bucket")
+
+	return cmd
+}
+
+func VeleroConfigureAmazonS3Cmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "configure-aws-s3",
+		Short:         "Configure snapshots to use AWS S3 storage",
+		Long:          ``,
+		SilenceUsage:  true,
+		SilenceErrors: false,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			viper.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			v := viper.GetViper()
+
+			namespace := v.GetString("namespace")
+			if err := validateNamespace(namespace); err != nil {
+				return err
+			}
+
+			clientset, err := k8sutil.GetClientset()
+			if err != nil {
+				return errors.Wrap(err, "failed to get clientset")
+			}
+
+			veleroStatus, err := snapshot.DetectVelero(cmd.Context(), namespace)
+			if err != nil {
+				return errors.Wrap(err, "failed to detect velero")
+			}
+			if veleroStatus == nil {
+				return errors.New("velero namespace not found")
+			}
+
+			if !veleroStatus.ContainsPlugin("velero-plugin-for-aws") {
+				return errors.New("velero does not have the 'velero-plugin-for-aws' installed; " +
+					"consult https://kots.io/kotsadm/snapshots/overview/ for install instructions`)")
+			}
+
+			if v.GetString("bucket") == "" {
```

Comment:
how about `MarkFlagRequired`?

Reply:
Yeah, I thought about that but turns out it's just for shell completion. I decided not to add this since I anticipate the shell completion won't play well with a krew plugin. Let me know if I'm mistaken though on either count.

Reply:
Updated per our slack convo!